functions.php

```diff
@@ -242,15 +244,29 @@ function getCoordinatesForAddress($address) {
     $coordinates = new Coordinates();
 
     if (strlen($address) > 0) {
-        $map_details_response = get($GLOBALS['maps_endpoint']
-            . "&postalcode="
-            . urlencode($address)
-            . "&components=" . urlencode(setting('location_lookup_bias')));
-        $map_details = json_decode($map_details_response);
-        if ($map_details[0]->place_id != null) { 
-            $coordinates->location  = $map_details[0]->display_name; 
-            $coordinates->latitude  = $map_details[0]->lat; 
-            $coordinates->longitude = $map_details[0]->lon; 
+        if (isset($google_maps_api_key)) {
+                $map_details_response = get($GLOBALS['google_maps_endpoint']
+                . "&address="
+                . urlencode($address)
+                . "&components=" . urlencode(setting('location_lookup_bias')));
+            $map_details = json_decode($map_details_response);
+            if (count($map_details->results) > 0) {
+                $coordinates->location  = $map_details->results[0]->formatted_address;
+                $geometry               = $map_details->results[0]->geometry->location;
+                $coordinates->latitude  = $geometry->lat;
+                $coordinates->longitude = $geometry->lng;
+            }
+        } else {
+            $map_details_response = get($GLOBALS['osm_maps_endpoint']
+                . "&postalcode="
+                . urlencode($address)
+                . (isset($GLOBALS['location_lookup_bias']) ? "&country=" . urlencode($GLOBALS['location_lookup_bias']) : ""));
```

Comment:
this should probably be "&countrycodes=" using the ISO 3166-1alpha2 code, per nominatim spec it will honor this, but will not always with just &country